lib/benchee/statistics/percentile.ex

```diff
@@ -1,50 +1,67 @@
 defmodule Benchee.Statistics.Percentile do
   @moduledoc false
 
+  @type percentile :: float()
+  @type percentiles :: %{optional(number()) => percentile()}
+
   @doc """
-  Calculates the value at the `percentile_number`-th percentile. Think of this as the
-  value below which `percentile_number` percent of the samples lie. For example,
+  Calculates the value at the `percentile_rank`-th percentile. Think of this as the
+  value below which `percentile_rank` percent of the samples lie. For example,
   if `Benchee.Statistics.Percentile.percentile(samples, 99)` == 123.45,
   99% of samples are less than 123.45.
 
-  ## Examples
-
-  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, 100)
-  5.0
-
-  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, 150)
-  5.0
+  Passing a number for `percentile_rank` calculates a single percentile.
+  Passing a list of numbers calculates multiple percentiles, and returns them
+  as a map like %{90 => 45.6, 99 => 78.9}, where the keys are the percentile
+  numbers, and the values are the percentile values.
 
-  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, 0)
-  1.0
+  ## Examples
 
-  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, -1)
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 12.5)
   1.0
 
   iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 50)
   3.0
 
   iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 75)
   4.75
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 99)
+  5.0
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], [50, 75, 99])
+  %{50 => 3.0, 75 => 4.75, 99 => 5.0}
```

Comment:
I feel a bit uneasy about having a different return type for one argument versus list arguments. Not saying needs to change just expressing fears. I think from a user's perspective this here is usually nicer but can come unexpected

Reply:
Good point. What do  you think about renaming this function `percentiles/2`, always passing in a list of percentiles, and always returning a map. I kind of like that idea.

Reply:
Sounds good to me :+1: